src/aleph_vrf/coordinator/vrf.py

```diff
@@ -336,7 +336,7 @@ async def get_existing_vrf_message(
     if messages.messages:
         if len(messages.messages) > 1:
             logger.warning(f"Multiple VRF messages found for request id {request_id}")
-        return messages.messages[0]
+        return messages.messages[-1]
```

Comment:
Why not force the order in `get_messages` to ensure that they are sorted adequately ? And why not add a comment on why you need to get the `[-1]` ?